src/app/public/src/modules/action-buttons/action-buttons.component.ts

```diff
@@ -1,13 +1,42 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 
 import { StacheNav, StacheNavLink, StacheNavComponent } from '../nav';
+import { InputConverter } from '../shared';
 
 @Component({
   selector: 'stache-action-buttons',
   templateUrl: './action-buttons.component.html',
   styleUrls: ['./action-buttons.component.scss']
 })
-export class StacheActionButtonsComponent extends StacheNavComponent implements StacheNav {
+export class StacheActionButtonsComponent extends StacheNavComponent implements StacheNav, OnInit {
   @Input()
   public routes: StacheNavLink[];
+
+  @Input()
+  @InputConverter()
+  public showSearch: boolean = true;
+
+  private originalRoutes: StacheNavLink[];
+
+  public ngOnInit() {
+    this.originalRoutes = this.routes.slice();
+  }
+
+  public searchApplied(searchText: string) {
+    let filteredItems = this.originalRoutes;
+    if (searchText) {
+      filteredItems = this.routes.filter(function (item: any){
+        let property: any;
+        for (property in item) {
+          if (item.hasOwnProperty(property) && (property === 'name' || property === 'summary')) {
+            if (item[property].toLowerCase().indexOf(searchText.toLowerCase()) > -1) {
+              return true;
+            }
+          }
+        }
+        return false;
+      });
+    }
+    this.routes = filteredItems;
```

Comment:
We can simplify this,  don't change the this.routes, instead create a new `filteredRoutes`  we can iterate over in the `ngFor`, keeping the original routes in place.  This will also eliminate the need for a variable inside this , method.

```
  private filteredRoutes: StacheNavLink[];

  public ngOnInit() {
    this.filteredRoutes = this.routes.slice();
  }

  public searchApplied(searchText: string) {
    this.filteredRoutes = this.routes;
    if (searchText) {
      this.filteredRoutes = this.routes.filter(function (item: any){
        let property: any;
        for (property in item) {
          if (item.hasOwnProperty(property) && (property === 'name' || property === 'summary')) {
            if (item[property].toLowerCase().indexOf(searchText.toLowerCase()) > -1) {
              return true;
            }
          }
        }
        return false;
      });
    }
  }
```